presto-hive/src/main/java/com/facebook/presto/hive/authentication/UserGroupInformationUtils.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive.authentication;
+
+import org.apache.hadoop.security.UserGroupInformation;
+
+import java.security.PrivilegedAction;
+import java.util.concurrent.atomic.AtomicReference;
+
+final class UserGroupInformationUtils
```

Comment:
I think it is better to keep here as much as possible. Moving the code to the shaded library requires copy paste of it to the 3 distinct repositories. I would rather create only the thin shims in shaded repositories.